cabal-install/Distribution/Client/ProjectConfig/Types.hs

```diff
@@ -309,7 +309,8 @@ data SolverSettings
        solverSettingLocalRepos        :: [FilePath],
        solverSettingConstraints       :: [(UserConstraint, ConstraintSource)],
        solverSettingPreferences       :: [Dependency],
-       solverSettingFlagAssignment    :: FlagAssignment, --TODO: [required eventually] must be per-package, not global
+       solverSettingFlagAssignment    :: FlagAssignment, --TODO: [required eventually] eliminate this global one
```

Comment:
So why is this still here? A little more detail why this couldn't straightforwardly eliminated now will someone looking at this code later.